lib/goo/sparql/solutions_mapper.rb

```diff
@@ -157,35 +164,56 @@ def get_value_object(id, objects_new, object, list_attributes, predicate)
         end
 
         if list_attributes.include?(predicate)
-          # To handle attr that are lists
-          pre = if @klass_struct
-                  @models_by_id[id][predicate]
-                else
-                  @models_by_id[id].instance_variable_get("@#{predicate}")
-                end
-          if object.nil? && pre.nil?
-            object = []
-          elsif object.nil? && !pre.nil?
-            object = pre
-          elsif object
-            object = !pre ? [object] : (pre.dup << object)
-            object.uniq!
-          end
+          pre = @klass_struct ? @models_by_id[id][predicate] : @models_by_id[id].instance_variable_get("@#{predicate}")
+
+          object = []  if object.nil? && pre.nil?
+
+          object = pre if object.nil? && !pre.nil?
+
+          object = pre.nil? ? [object] : (pre.dup << object)
+          object.uniq
+
         end
         [object, objects_new]
       end
 
-      def add_object_to_model(id, object, predicate, lang)
+      def object_language(new_value)
+        new_value.language || :no_lang if new_value.is_a?(RDF::Literal)
+      end
+
+      def language_match?(language, requested_lang = nil)
+        !language.nil? && (language.eql?(requested_lang) || language.eql?(:no_lang) || requested_lang.nil?)
+      end
+
+      def add_object_to_model(id, objects, current_obj, predicate, language, requested_lang = nil)
         if @models_by_id[id].respond_to?(:klass)
-          @models_by_id[id][predicate] = object unless object.nil? && !@models_by_id[id][predicate].nil?
+          @models_by_id[id][predicate] = objects unless objects.nil? && !@models_by_id[id][predicate].nil?
         elsif !@models_by_id[id].class.handler?(predicate) &&
-              !(object.nil? && !@models_by_id[id].instance_variable_get("@#{predicate}").nil?) &&
+              !(objects.nil? && !@models_by_id[id].instance_variable_get("@#{predicate}").nil?) &&
               predicate != :id
 
-          if (lang&.eql?(:no_lang)) || !lang
-            @models_by_id[id].send("#{predicate}=", object, on_load: true)
+          if language.nil? # the object is a non-literal
```

Comment:
create a method `no_literal?` instead of the comment 

```ruby 
def no_literal?(lang)
   lang.nil?
end
```

Reply:
it's not a reusable function , and no_literal should depend on the object not the lang